script.js

```diff
@@ -1,52 +1,79 @@
-const container = document.getElementById("container");
-const id = Date. now();
-
-const addToListButton = document.getElementById('add-lists');
-const deleteItemTag = 'span';
-const inputItemTag = 'input';
-
-const saveData = () => {
-  localStorage.setItem('data', container.innerHTML);
+/**  
+ * @type {{ 
+*  items: [
+  *    [
+  *      { 
+  *        text: String, 
+  *        isDone: Boolean 
+  *      }, 
+  *      {
+  *        text: String, 
+ *         isDone: Boolean 
+  *      },
+  *      ...
+  *    ],
+  *    []
+  *  ] 
+  * }}
+  */
+
+const rootState = [];
+const rootContainer = document.getElementById("container");
+
+const notifyStateUpdated = () => {
+  localStorage.setItem('data', rootContainer.innerHTML);
```

Comment:
We need to save `rootState` instead of html, bc otherwise our UI doesn't work after restoring.
Keep it in mind that local storage accepts only strings, so you need to stringify your state